nifi-extension-bundles/nifi-standard-bundle/nifi-standard-rules/pom.xml

```diff
@@ -23,7 +23,13 @@
     <artifactId>nifi-standard-rules</artifactId>
     <packaging>jar</packaging>
 
-    <dependencies />
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-framework-api</artifactId>
+            <scope>provided</scope>
```

Comment:
Don't need to specify the scope here, because nifi-bom declares it as provided.

Reply:
Removed.

Reply:
I looked into this, too, @exceptionfactory.  I found that the org.apache.nifi.controller.queue.LoadBalanceStrategy and org.apache.nifi.controller.queue.LoadBalanceCompression classes used in this rule are located in nifi-framework-api.

Reply:
@mosermw, although those classes are available at that level, general usage is restricted to components that implement framework-level interfaces. This seems like a pattern to avoid for Flow Analysis Rules, as these implement `nifi-api` interfaces as opposed to `nifi-framework-api` interfaces.

Reply:
This code has to check the return value of connection.getLoadBalanceStrategy() against configured strategies to see if it would be allowed/disallowed.  An alternative to referencing nifi-framework-api is to hard-code those strings directly in this rule.

I suspect the LoadBalanceStrategy and LoadBalanceCompression classes belong in nifi-api.  But that's a separate discussion ;).

Reply:
The reason is helpful, but the concern is not the dependency size, as it is provided, but with the logical distinction between framework components and extension components. Under the circumstances, using String values instead of adding the dependency would be the best way to go.

Reply:
@exceptionfactory I removed the `nifi-framework-api` dependency.